DigiMeSDK/Core/Classes/Security/DMECrypto.h

```diff
@@ -52,6 +52,42 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (nullable NSData *)decryptAes256UsingKey:(NSData *)keyData initializationVector:(NSData *)ivData data:(NSData *)data error:(NSError * __autoreleasing *)error;
 
+/**
+ Generates random data using length as a parameter.
+ 
+ @param length int
+ @return NSData - rundom bytes for the specified length.
+ */
+- (NSData *)getRandomUnsignedCharacters:(int)length;
+
+/**
+ Encrypt metadata for Postbox with AES encryption
+ 
+ @param symmetricalKey NSData
+ @param iv NSData
+ @param metadata NSData
+ @return NSString - AES encrypted metadata to push to postbox in Base64 encoding.
+ */
+- (NSString *)preparePostboxMetadataWithKey:(NSData *)symmetricalKey initializationVector:(NSData *)iv metadata:(NSData *)metadata;
+
+/**
+ Encrypt data for Postbox with AES encryption
+ 
+ @param symmetricalKey NSData
+ @param iv NSData
+ @param data NSData
+ @return NSString - AES encrypted data to push to postbox in a hexadecimal representation.
+ */
+- (NSData *)preparePostboxDataWithKey:(NSData *)symmetricalKey initializationVector:(NSData *)iv dataToPush:(NSData *)data;
+
+/**
+ Encrypt Symmetrical Key for Postbox with RSA public key and return it as Base64 encoded.
+ 
+ @param dataToEncrypt NSData
+ @param publicKey NSString
+ */
+- (NSString *)preparePostboxSymmetricalKeyWithData:(NSData *)dataToEncrypt rsaPublicKeyForEncryption:(NSString *)publicKey;
```

Comment:
I think a made a comment above about this, but just to reiterate here, the method name does not really match description.

How about:

```objective-c
/**
 Encrypt Symmetrical Key with RSA public key and return it as Base64 encoded.
 
 @param symmetricalKey NSData
 @param rsaPublicKey NSString
 */
- (NSString *)encryptSymmetricalKey:(NSData *)symmetricalKey rsaPublicKey:(NSString *)publicKey;
```